STM32WB Series support

erwango opened the pull request:
Add support for dual core, multi protocol wireless STM32 controllers.
Provide support for nucleo_wb55rg BLE target

erwango committed:
hal: stm32wbxx: Add HAL for the STM32WB series

This adds the HAL code needed for the SoC for the stm32wb series.

Origin: ST Microelectronics
License: BSD-3-Clause
URL: http://www.st.com/en/embedded-software/stm32cubewb.html
Commit: 1.0.0
Purpose: HAL layer for stm32wb
Maintained-by: External

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
ext/hal/st: Adapt stm32cube folder to stm32wb introduction

Following introduction of STM32Cube for STM32WB, update stm32cube
folder to enable build of STM32WB Cube package.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
soc/arm/st_stm32: Add support for stm32wb soc

Provide basic soc configuration for STM32WB SoCs support in Zephyr.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: Add support to stm32wb series

Add support to stm32wb series in stm32 clock_control driver.
Ip is similar to stm32l4 one but AHB bus presacler is renamed
to "CPU1" and CPU2 and AHB4 prescalers should be defined.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/interrupt_controller: stm32: Add support for stm32wb series

Update exti driver to support STM32WB series. IP is similar
to stm32l4.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
soc/arm/st_stm32: stm32wb: Add gpio support

Add GPIO support to stm32wb series.
Only ABCDE and H ports are available for now on this series.
Accordingly, update series dtsi file.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/pinmux: stm32: Add pinmux definitions for stm32wb (LP)U(S)ART

Add definitions for LPUART1 and USART1.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

dbkinder commented on boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst:
```diff
@@ -0,0 +1,267 @@
+- On-board ST-LINK/V2-1 debugger/programmer with USB re- enumeration capability:
+  mass storage, virtual COM port and debug port
+
+.. image:: img/nucleowb55rg.jpg
```
The board image should be in ``boards/arm/nucleo_wb55rg/doc/img/nucleowb55rg.jpg`` and you've got it in ``boards/arm/nucleo_wb55rg/img/nucleowb55rg.jpg``

dbkinder commented on boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst:
```diff
@@ -0,0 +1,267 @@
+.. _Nucleo WB55RG website:
+   https://www.st.com/en/evaluation-tools/p-nucleo-wb55.html
+
+.. _STM32WB55RGG on www.st.com:
```
Looks like you've got an extra ``G`` on this label (hence the doc build error)

dbkinder reviewed (changes requested):
Looks good, other than the two doc build errors...

dbkinder reviewed (approved):
+1 for docs

therealprof commented on boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst:
```diff
@@ -0,0 +1,267 @@
+********
+
+STM32WB55RG is an ultra-low-power dual core Arm Cortex-M4 MCU 64 MHz,Cortex-M0 32MHz
+with 1 Mbyte of Flash memory, Bluetooth 5, 802.15.4, USB, LCD, AES-259 SoC and
```
```suggestion
with 1 Mbyte of Flash memory, Bluetooth 5, 802.15.4, USB, LCD, AES-256 SoC and
```

erwango replied:
@therealprof, txs for the catch. And while you're here, would you mind check my last comment on our pending discussion (https://github.com/zephyrproject-rtos/zephyr/pull/9300#discussion_r239447973)? Would be nice, txs.

erwango replied:
fixed

therealprof commented on boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst:
```diff
@@ -0,0 +1,267 @@
+- Development support: serial wire debug (SWD), JTAG, Embedded Trace Macrocell |trade|
+
+
+More information about STM32L496ZG can be found here:
```
STM32L496ZG?

therealprof commented on boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst:
```diff
@@ -0,0 +1,267 @@
+Supported Features
+==================
+
+The Zephyr nucleo_l496zg board configuration supports the following hardware features:
```
nucleo_l496zg?

therealprof commented on boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst:
```diff
@@ -0,0 +1,267 @@
+in the drive mounted when plugging your nucleo board to your PC.
+
+
+Flashing an application to Nucleo L496ZG
```
dito

erwango replied:
dito

erwango committed:
boards: Add support for nulceo_wb55rg

Basic support for nucleo_wb55rg board.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
dts/arm/st: Add serial nodes on stm32wb

Add USART1 and LPUART1 nodes on stm32wb series.
Only these 2 ones are available for now on this series.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak commented on dts/arm/st/wb/stm32wb.dtsi:
```diff
@@ -0,0 +1,47 @@
+				label = "FLASH_STM32";
+			};
+		};
+
```
nit remove blank line.

erwango replied:
Txs! Fixed now!

galak reviewed (changes requested):
One minor nit, otherwise looks good.

erwango committed:
dts/arm/st: Provide basis for stm32wb device tree

Base .dtsi files for stm32wb series and stm32wb55 SoC.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
